source/_docs/private-paths.md

```diff
@@ -1,87 +1,99 @@
+```
+if (isset($_ENV['PANTHEON_ENVIRONMENT']) && $_ENV['PANTHEON_ENVIRONMENT'] == 'live') {
+  $json_text = file_get_contents('sites/default/files/private/stripe_live.json');
+  $stripe_data = json_decode($json_text, TRUE);
+  $conf['stripe_key'] = $stripe_data['key'];
+}
+else {
+  // We aren't in prod, load a fallback or null key.
+  $conf['stripe_key'] = 'foo';
+}
+```
+This Drupal example reads the key from the private file `stripe_live.json` only when the request is made from the Live environment on Pantheon.
+
+### Additional Drupal Configuration
+
+These files will be web-accessible based on the access control rules that you set for your site and will use the following directory: `sites/default/files/private`
+
+To configure, go to **Administration** > **Configuration** > **Media** > **File System**, select **Private local files served by Drupal** as the default download method, and click **Save Configuration**.
+
+
+## Troubleshooting
+
+### Resolving Warning: file_put_contents(private:///.htaccess)
+
+If you receive the error above, make sure that the private path for code or files exists in your repository. If you are configuring a private path for code, you'll need to start from your Dev environment and create the private directory, then commit via Git, or create via SFTP and commit via Pantheon Dashboard. Once the directory has been created and committed, you can deploy to Test and Live to deploy the new directory. Once the directory exists you can resubmit your changes via the file systems settings page in your Drupal Admin interface for each environment.
+
+### Selectively Exposing Code
+
+If you have a private code library that needs to have a specific sub-directory exposed (e.g. using SimpleSamlPHP), you can do this with symlinks:
+
+    # from within a git checkout
+    ln -s private/simplesamlphp/www ./simplesaml
+    git add simplesaml
+    git commit simplesaml -m "adding simplesaml symlink"
+    git push origin master
+
+The result will be a web-accessible URL at https://dev.yoursite.pantheonsite.io/simplesaml which will point to the code in `/private/simplesamlphp/www`.
+
+### Setting Commerce Kickstart or Ubercart Key Path
+
+Make sure to set a relative path. This ensures the key path will work on all appservers across the site's environments.
+You can either set the path in the Drupal admin interface or with Terminus and Drush:
+
+1. Set the encryption key path:
+
+   ```bash
+   terminus drush <site>.<env> -- vset uc_credit_encryption_path <my_private_path>
+   ```
+
+   `<my_private_path>` can be set to either of these non-web accessible private directories:
+
+    - `'sites/default/files/private'` (preferred)
+    - `'private'` (version controlled)
+
+
+2. Verify that `uc_credit_encryption_path` is set correctly:
+
+   ```bash
+   terminus drush <site>.<env> -- vget uc_credit_encryption_path
+   ```
+
+3. Check that the output of `uc_credit_encryption_path:` matches your settings.
```

Comment:
is `uc_credit_encryption_path:` a shell command?

Reply:
`uc_credit_encryption_path` is a code level variable, set in the database.

The drush vset command sets the variable:
`terminus drush <site>.<env> -- vset uc_credit_encryption_path`

The drush vget command checks what the variable is currently set to:
`terminus drush <site>.<env> -- vget uc_credit_encryption_path`